scrapy_poet/page_input_providers.py

```diff
@@ -101,6 +106,26 @@ def __init__(self, crawler: Crawler):
         """Initializes the provider. Invoked only at spider start up."""
         pass
 
+    def fingerprint(self, to_provide: Set[Callable], request: Request) -> str:
```

Comment:
I wonder if those 3 methods can be moved to an abstract mixing. When you implement a provider which supports caching, you inherit from this mixin, and get all your methods checked (that all 3 needs to be implemented).

Reply:
Good call. Refactored this on https://github.com/scrapinghub/scrapy-poet/pull/55/commits/15c5a7fe4fd968c2cb103ffb9820b79791dc9119.

Made it into an abstract base class to force the user to implement them when subclassing. Also fixed a bug where if `SCRAPY_POET_CACHE` is being used, all providers should support the cache. This should not be the case as each provider can optionally support it.